strassengezwitscher/strassengezwitscher/models.py

```diff
@@ -1,7 +1,30 @@
 from __future__ import unicode_literals
-
+from decimal import Decimal
 from django.db import models
 from django.utils.encoding import python_2_unicode_compatible
+from rest_framework.exceptions import ValidationError
+from rest_framework import filters
+
+
+class MapObjectFilter(filters.BaseFilterBackend):
+
+    def filter_queryset(self, request, queryset, view):
+
+        square_params = {
+            'min_lat': request.query_params.get('min_lat', None),
+            'min_long': request.query_params.get('min_long', None),
+            'max_lat': request.query_params.get('max_lat', None),
+            'max_long': request.query_params.get('max_long', None)
+        }
+
+        if square_params['min_lat'] is not None or \
```

Comment:
Die Idee hier ist die folgende: 
Falls einer der Parameter gesetzt ist, soll offensichtlich gefiltert werden. Das Validieren der Filter-Parameter und das Filtern an sich wird im Model ausgeführt, weil dort ersichtlich wird, warum welche Parameter wie gesetzt sein müssen.
Falls keiner der Parameter gesetzt ist, soll nicht gefiltert werden und es werden alle Objekte zurück gegeben. 

Bei deiner Variante müssten mehrere Fällte unterschieden werden:

1. Sind alle parameter gesetzt
Falls ja, muss selbst dann noch im Model gecheckt werden, ob es sich um Strings handelt, die in Decimals umgewandelt werden können und ob sie die Anforderungen an Longitude/Latitude erfüllen und ansonsten ein Validation Error geworfen werden

2. Sind ein Teil der Parameter gesetzt
--> dann muss hier ein Validation Error geworfen werden (Dieser Fall wird in meiner Variante abgedeckt in der Methode „get_as_decimals“ im model)

3. Sind keine Parameter gesetzt
--> dann werden alle objekte zurück gegeben

Ich finde meine Variante daher besser.